src/managers/AdManager/AdManager.ts

```diff
@@ -0,0 +1,129 @@
+import { MAX_TIMEOUT_MS, SLOT_NAME, TPL_CODE } from 'consts/dlApi';
+import {
+  ERROR_PROMOTED_PRODUCTS_MSG,
+  LINK_SELECTOR_DOES_NOT_MATCH,
+  REQUEST_TIMED_OUT,
+} from 'consts/messages';
+import { VIEWS_MAP } from 'consts/views';
+import { getProductsIds } from 'managers/AdManager/AdManager.util';
+import { TPages } from 'types/pages';
+import { TAdProduct, TFormattedProduct } from 'types/product';
+import getMessage from 'utils/getMessage';
+import getProductsCountToInject from 'utils/getProductCountToInject';
+
+class AdManager {
+  private page: TPages;
+  private productsIds: number[];
+
+  constructor(page: TPages) {
+    this.page = page;
+
+    if (this.page) {
+      this.productsIds = getProductsIds();
+    }
+  }
+
+  public getPromotedProducts = async () => {
+    if (!dlApi.fetchNativeAd)
+      throw new Error(getMessage(ERROR_PROMOTED_PRODUCTS_MSG));
+
+    let products: TAdProduct[] | null | Error = null;
+
+    const timeoutPromise = new Promise<void>((_, reject) => {
+      setTimeout(() => {
+        reject(getMessage(REQUEST_TIMED_OUT));
+      }, MAX_TIMEOUT_MS);
+    });
+
+    const fetchNativeAd = new Promise<TAdProduct[]>((resolve, reject) => {
+      dlApi.cmd = dlApi.cmd || [];
+      dlApi.cmd.push(async (dlApiObj) => {
+        try {
+          const ads = await dlApiObj.fetchNativeAd!({
+            slot: SLOT_NAME,
+            opts: {
+              offer_ids: this.productsIds.join(','),
+            },
+            tplCode: TPL_CODE,
```

Comment:
Wywołanie funkcji fetchNativeAd powinno przyjmować jeszcze jeden argument - właściwość "div".
Tak, jak w przypadku IdoSell.
Jest to unikalny identyfikator diva, w którym znajduje się oferta. Ważne, aby ten div obejmował cały element z ofertą sponsorowaną.